### ISSUE TITLE
GuildMemberManager.fetch is not reentrant

### ISSUE THREAD

**Issue by ikelly** [discordjs/discord.js#4883]
When calling guild.members.fetch({user: userIds}) a second time before the first call finishes, both Promises return the results of the first fetch.

**Comment by wasdennnoch** [comment]
It is intended for `GuildMemberManager.fetch()` to cache the resulting user, and return any cached user, if found. If you wish to force-fetch the user from the API, use the `force: true` argument in the argument object.

If this isn't what you mean, please be more precise and include an example code snippet. Also please don't just delete the issue template - it's there for a reason.


Actually I've just dove through the code a bit and found that if you call `fetch` twice within the same millisecond, the request will use the same nonce internalls and thus the second query will return with the results of the first one, even when using different arguments for both queries. Is this what you are doing?
https://github.com/discordjs/discord.js/blob/d2341654fe6ef87c6048bb5bf74c394062f3f618/src/managers/GuildMemberManager.js#L266
If so, does the issue still occurs when you supply a more random `nonce` in your fetch arguments?

**Comment by ikelly** [comment]
Sorry, I'm not sure what happened but somehow there was no template when I created the issue.

To clarify, here's a sample code snippet

```
let promise1 = guild.members.fetch({user: userIds1});
let promise2 = guild.members.fetch({user: userIds2});
```

The two sets of user IDs are different, so I expect different results, but when they resolve both Promises contain the results of the first fetch. I'm not concerned with whether the results were returned from cache or not.

I don't know whether the calls were happening during the same millisecond, but it seems plausible since the issue was not 100% reproducible. I just tried passing unique nonces as you suggest and I wasn't able to replicate the issue when doing that.
